File: api-tester-java/src/main/java/com/revenuecat/apitesterjava/PurchaseStateAPI.java
package com.revenuecat.apitesterjava;

import com.revenuecat.purchases.models.PurchaseState;

@SuppressWarnings({"unused"})
final class PurchaseStateAPI {
    static void check(final PurchaseState state) {
        switch (state) {
            case UNSPECIFIED_STATE:
            case PURCHASED:
            case PENDING:
        }
    }
}

File: api-tester-kotlin/src/main/java/com/revenuecat/apitesterkotlin/PurchaseStateAPI.kt
package com.revenuecat.apitesterkotlin

import com.revenuecat.purchases.models.PurchaseState

@Suppress("unused")
private class PurchaseStateAPI {
    fun check(state: PurchaseState) {
        when (state) {
            PurchaseState.UNSPECIFIED_STATE,
            PurchaseState.PURCHASED,
            PurchaseState.PENDING
            -> {}
        }
    }
}
